test/system/src/test/java/io/pravega/test/system/MultiReaderTxnWriterWithFailoverTest.java

```diff
@@ -0,0 +1,468 @@
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+        //seal the stream
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(scope, STREAM_NAME);
+        log.info("Sealing stream {}", STREAM_NAME);
+        assertTrue(sealStreamStatus.get());
+        //delete the stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(scope, STREAM_NAME);
+        log.info("Deleting stream {}", STREAM_NAME);
+        assertTrue(deleteStreamStatus.get());
+
+        //delete the scope
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(scope);
+        log.info("Deleting scope {}", scope);
+        assertTrue(deleteScopeStatus.get());
+        log.info("Test {} succeeds ", "MultiReaderWriterTxnWithFailOver");
+    }
+
+    private void performFailoverTest() throws InterruptedException {
+
+        long currentWriteCount1;
+        long currentReadCount1;
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+
+        currentWriteCount1 = eventWriteCount.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Read count: {}, write count: {} without any failover", currentReadCount1, currentWriteCount1);
+
+        int sleepTime;
+
+        //check reads and writes after some random time
+        sleepTime = new Random().nextInt(50000) + 3000;
+        log.info("Sleeping for {} ", sleepTime);
+        Thread.sleep(sleepTime);
+
+        long currentWriteCount2;
+        long currentReadCount2;
+
+        currentWriteCount2 = eventWriteCount.get();
+        currentReadCount2 = eventReadCount.get();
+
+        log.info("Read count: {}, write count: {} without any failover after sleep before scaling", currentReadCount2, currentWriteCount2);
+
+        //ensure writes are happening
+        assertTrue(currentWriteCount2 > currentWriteCount1);
+        //ensure reads are happening
+        assertTrue(currentReadCount2 > currentReadCount1);
+
+        //Scale down SSS instances to 2
+        segmentStoreInstance.scaleService(2, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
+        Thread.sleep(ZK_DEFAULT_SESSION_TIMEOUT);
+        log.info("Scaling down SSS instances from 3 to 2");
+
+        currentWriteCount1 = eventWriteCount.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Read count: {}, write count: {} after SSS failover after sleep", currentReadCount1, currentWriteCount1);
+
+        //ensure writes are happening
+        assertTrue(currentWriteCount1 > currentWriteCount2);
+        //ensure reads are happening
+        assertTrue(currentReadCount1 > currentReadCount2);
+
+        //Scale down controller instances to 2
+        controllerInstance.scaleService(2, true);
+        //zookeeper will take about 60 seconds to detect that the node has gone down
```

Comment:
Yeah we are relying on the the curator client's default for this (I had miss-communicated this earlier; the zookeeper deployment configuration is for the min and max values based on its tickTime configuration). We need to expose the session timeout value via segment store's and controller's configuration (this is where we register to zookeeper) if we want to control the session timeout; or we can just use a lower default for the session timeout if that makes sense for our deployments. @fpj any suggestions?